From 7d92d9d1d3705c164cbbb83b26c0b404efcafdce Mon Sep 17 00:00:00 2001
From: Kevin Wang <kevin1.wang@amd.com>
Date: Tue, 3 Dec 2019 19:59:55 +0800
Subject: [PATCH 4638/4736] Revert "drm/amdgpu: Set GTT size to be bigger than
 3/4 of RAM"

This reverts commit 55dc9b6a685260f4545a0f16c16bec9756b71a86.

This commit will cause oom issue when do vulcan CTS Test.

Signed-off-by:Kevin Wang <kevin1.wang@amd.com>
Acked-by: Feifei Xu <feifei.xue@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
index d93bfaca5daf..6d295e7ee444 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
@@ -2156,10 +2156,11 @@ int amdgpu_ttm_init(struct amdgpu_device *adev)
 		struct sysinfo si;
 
 		si_meminfo(&si);
-		gtt_size = max3((AMDGPU_DEFAULT_GTT_SIZE_MB << 20),
-				adev->gmc.mc_vram_size,
-				((uint64_t)si.totalram * si.mem_unit));
-	} else
+		gtt_size = min(max((AMDGPU_DEFAULT_GTT_SIZE_MB << 20),
+			       adev->gmc.mc_vram_size),
+			       ((uint64_t)si.totalram * si.mem_unit * 3/4));
+	}
+	else
 		gtt_size = (uint64_t)amdgpu_gtt_size << 20;
 
 	/* reserve for DGMA import domain */
-- 
2.17.1

